src/main/java/com/dataguise/saas/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,35 @@
+package com.dataguise.saas;
+
+import com.dataguise.saas.service.JWTService;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class JWTAuthenticationFilter extends GenericFilterBean {
+
+    @Override
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.design.DesignForExtensionCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/JWTAuthenticationFilter.java L18
    
    Class &#39;JWTAuthenticationFilter&#39; looks like designed for extension (can be subclassed), but the method &#39;doFilter&#39; does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class &#39;JWTAuthenticationFilter&#39; final or making the method &#39;doFilter&#39; static/final/abstract/empty, or adding allowed annotation for the method.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1794327312>*